kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -560,6 +574,67 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ColumnVector visitLike(final Like like) {
+            PredicateChildrenEvalResult argResults = evalBinaryExpressionChildren(like);
+            ColumnVector left = argResults.leftResult;
+            ColumnVector right = argResults.rightResult;
+            int numRows = argResults.rowCount;
+            boolean[] result = new boolean[numRows];
+            boolean[] nullability = new boolean[numRows];
+            for (int rowId = 0; rowId < numRows; rowId++) {
+                String leftOp = left.getString(rowId);
+                String rightOp = right.getString(rowId);
+                if (!Objects.isNull(leftOp) && !Objects.isNull(rightOp)) {
+                    nullability[rowId] = false;
+                    String rightOpRegexPattern =
+                            escapeLikeRegex(rightOp, like.getEscape());
```

Comment:
optimization: can be in a follow up PR. We can check the child to see if it is a literal (i.e that pattern is constant), we can get the regex once and not for every row.

Reply:
sure. will create a git issue for this.

Reply:
issue for performance fix - https://github.com/delta-io/delta/issues/3129